messagequeue/messagequeue.go

```diff
@@ -0,0 +1,216 @@
+package messagequeue
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	bsmsg "github.com/ipfs/go-bitswap/message"
+	bsnet "github.com/ipfs/go-bitswap/network"
+	wantlist "github.com/ipfs/go-bitswap/wantlist"
+	logging "github.com/ipfs/go-log"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+var log = logging.Logger("bitswap")
+
+type MessageQueue interface {
```

Comment:
Is there likely to be multiple implementations of the `MessageQueue` interface? If it is this is fine, but it seems like an unnecessary abstraction.

Reply:
I'm working towards at minimum a mock implementation for testing. One thing I'd like to get to is dependency injection patters for true unit tests. That's the main rationiale

Reply:
I am not the biggest fan of adding interfaces purely for testing/mock purposes, especially in go. A really good read on this is [Rakyll's blog on interface pollution](https://rakyll.org/interface-pollution/), also the [Go Wiki Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#interfaces).